auth/role.go

```diff
@@ -25,13 +25,14 @@ import (
 
 /** A group that users can belong to, with associated channel permissions. */
 type roleImpl struct {
-	Name_             string          `json:"name,omitempty"`
-	ExplicitChannels_ ch.TimedSet     `json:"admin_channels,omitempty"`
-	Channels_         ch.TimedSet     `json:"all_channels"`
-	Sequence_         uint64          `json:"sequence"`
-	ChannelHistory_   TimedSetHistory `json:"channel_history,omitempty"`   // Added to when a previously granted channel is revoked. Calculated inside of rebuildChannels.
-	ChannelInvalSeq   uint64          `json:"channel_inval_seq,omitempty"` // Sequence at which the channels were invalidated. Data remains in Channels_ for history calculation.
-	Deleted           bool            `json:"deleted,omitempty"`
+	Name_             string                                  `json:"name,omitempty"`
+	ExplicitChannels_ ch.TimedSet                             `json:"admin_channels,omitempty"`
+	Channels_         ch.TimedSet                             `json:"all_channels"`
```

Comment:
Should this become omit-empty? What do we do if there are only non default collections enabled?

Reply:
I left this as-is for now, because of the way we set the value to nil for channel invalidation.  The full implementation of invalidation for scopes and collections is still pending, could revisit at that time.